### ISSUE TITLE
Question at Cud forum

### ISSUE THREAD

**Issue by Alexey-T** [dinkumoil/cuda_shell_extension#8]
http://synwrite.sourceforge.net/forums/viewtopic.php?f=20&t=2349&start=10
@dinkumoil

**Comment by dinkumoil** [comment]
At first I want to repeat the user's posting in the forum:

> Hello,
>
> I have add parameters to user.json
> 
> Code: Select all
>     "ui_one_instance": true,
>     "ui_reopen_session": true,
> 
> 
> but it still open one tab when I open a file in explorer context.
> 
> How to keep one session permanently? Regardless when you open app with it *.exe file or open a file within context explorer?

Here the question is, did he try to open a file when an instance of _CudaText_ was already running or when there was no running instance of _CudaText_? What does he expect to happen in these two cases?

Furthermore, independently of the answers the user might give, I have to say that my shell extension can not **and will never** deal with CudaText session or other settings. The only thing the shell extension does is to run _CudaText_ and provide all the files selected in an _Explorer_ window as command line arguments. This is done for every packet of 125 selected files. That means, if the user selected 127 files the shell extension runs _CudaText_ twice, the first time with 125 files as command line arguments and the second time with the remaining 2 files.

What ever happens when _CudatText_ has been run by the shell extension is up to _CudaText_ and its active settings. If it is set to one-instance-mode the single-instance code (which I improved somewhere in the past) comes into play. The first _CudaText_ instance started simply takes the provided command line and processes it (this is code I haven't written and I don't know it). All subsequently started _CudaText_ instances "morse" the provided command line to the already running instance (the main instance) and terminate themselfs after that. The main instance then processes the received command line in a way I don't know and I haven't changed when I improved the single-instance code in the past.

Long story short, the user's problem mentioned above has nothing to do with my shell extension nor with my single-instance startup code.

**Comment by Alexey-T** [comment]
Thanks anyway
